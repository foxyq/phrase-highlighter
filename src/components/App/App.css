body {
  font-size: calc(10px + 2vmin);
}

.App {
  text-align: center;
  /* background-color: #282c34; */
  background-color: #ffffff;
  color: rgb(94, 92, 92);
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  font-size: calc(10px + 2vmin);
  /* color: white; */
}

.form {
  display: flex;
  flex-direction: column;
}

textarea {
  font-size: 0.7em;
}

input[type='submit'] {
  border: none;
  background-color: #45b5e6;
  border-radius: 5px;
  color: #fff;
  font-size: calc(10px + 2vmin);
  padding: 15px;
}

.highlight {
  position: relative;
  border-radius: 5px;
  border: 1px solid #fff;
}

.join-right {
  border-radius: 5px 0 0 5px;
  padding-right: 5px;
  margin-right: -5px;
}

.join-left {
  padding-left: 5px;
  margin-left: -5px;
}
