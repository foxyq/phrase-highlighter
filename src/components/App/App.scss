body {
  font-size: calc(10px + 2vmin);
}

.overlay {
  display: none;
  z-index: 2;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #000000ad;
}

.App {
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  color: rgb(94, 92, 92);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  line-height: 1.5;
  min-height: 100vh;
  position: relative;
  text-align: center;
  z-index: 1;
}

.form {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
}

section {
  width: 80vw;
}

textarea {
  font-size: 0.7em;
}

input[type='button'] {
  border: none;
  background-color: #45b5e6;
  border-radius: 5px;
  color: #fff;
  font-size: calc(10px + 2vmin);
  padding: 15px;
}

.highlight {
  cursor: pointer;
  padding: 2px;
  position: relative;
  padding-left: 8px;
  margin-left: -3px;
}

// #content .transparent-background {
//   background-color: transparent;
//   border-radius: 0;
// }

$colors: (
  'grey': #e8e8e8,
  'purple': #e1bbfc,
  'blue': #bfe6fc,
  'green': #e2f29a,
  'orange': #f1ce90
);

$zIndex: 0;

@each $color, $code in $colors {
  $zIndex: $zIndex + 1;

  .#{$color} {
    background-color: $code;
    z-index: $zIndex;

    &-transparent-background {
      background-color: transparent;
      border-radius: 0;
    }

    &-round-right {
      border-bottom-right-radius: 5px;
      border-top-right-radius: 5px;
      -webkit-box-shadow: 1px 0 0 0 #fff;
      box-shadow: 1px 0 0 0 #fff;
    }

    &-round-left {
      border-bottom-left-radius: 5px;
      border-top-left-radius: 5px;
      -webkit-box-shadow: 0 0 0 1px #fff;
      box-shadow: 0 0 0 1px #fff;
    }

    &-hide {
      background: transparent;
    }
  }
}

@each $color, $code in $colors {
  #content {
    .#{$color} {
      &-active {
        background-color: darken($code, 25%);
        color: white;
        -webkit-transition: all 0.1s ease-in-out;
        transition: all 0.1s ease-in-out;
        -webkit-box-shadow: none;
        box-shadow: none;

        &-round-right {
          border-bottom-right-radius: 5px;
          border-top-right-radius: 5px;
        }

        &-round-left {
          border-bottom-left-radius: 5px;
          border-top-left-radius: 5px;
        }
      }
    }
  }
}
