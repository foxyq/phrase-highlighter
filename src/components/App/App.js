import React, { Component } from 'react';
// import ReactDOM from 'react-dom';

import DynamicContent from '../DynamicContent';
import Form from '../Form';
import initState from '../../records/initialState';
import {
  inputIsEmpty,
  formatHighlights,
  createHighlightedElements
} from '../../utils';

import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = initState;
  }

  handleSubmit = e => {
    e.preventDefault();

    // check text
    const rawText = e.target.text.value;
    const formText = inputIsEmpty(rawText) ? this.state.text : rawText;

    const rawHighlights = e.target.highlights.value;
    const formHighlights = inputIsEmpty(rawHighlights)
      ? this.state.highlights
      : rawHighlights;

    // format highlights
    const newHighlights = formatHighlights(formHighlights);

    var reactElementDiv = createHighlightedElements(newHighlights, formText);

    // var reactElementDiv = createHighlightedElements(
    //   this.state.highlights,
    //   this.state.text
    // );

    this.setState({
      text: formText,
      highlights: newHighlights,
      toDisplay: reactElementDiv
    });
  };

  // handleTextChange = e => {
  //   const rawText = e.target.value;
  //   const formText = inputIsEmpty(rawText) ? this.state.text : rawText;
  //   // console.log(formText);
  //   this.setState({ text: formText });
  // };

  // handleHighlightsChange = e => {
  //   const rawHighlights = JSON.parse(e.target.value);

  //   console.log(rawHighlights);
  //   // const formHighlights = inputIsEmpty(rawHighlights)
  //   //   ? this.state.highlights
  //   //   : rawHighlights;

  //   // // format highlights
  //   // const newHighlights = formatHighlights(rawHighlights);

  //   this.setState({ highlights: rawHighlights });
  // };

  render() {
    return (
      <div className='App'>
        <section>
          <Form
            text={this.state.text}
            highlights={JSON.stringify(this.state.highlights)}
            handleSubmit={this.handleSubmit}
          />
        </section>
        <section>
          <DynamicContent>{this.state.toDisplay}</DynamicContent>
        </section>
      </div>
    );
  }
}

export default App;
