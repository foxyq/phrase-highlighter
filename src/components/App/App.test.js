import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

import initState from '../../records/initialState';

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const Component = renderer.create(<App />).toJSON();
    expect(Component).toMatchSnapshot();
  });
});
describe('handleSubmit behavior', () => {
  it('prevents default', () => {
    const Component = renderer.create(<App />);
    const preventDefault = jest.fn();
    const event = {
      preventDefault,
      target: {
        highlights: {
          value: 'Foo'
        },
        text: {
          value: 'Bar'
        }
      }
    };

    Component.getInstance().handleSubmit(event);

    expect(preventDefault).toBeCalled();
  });

  it('preserves state when given empty input', () => {
    const Component = renderer.create(<App />);
    const preventDefault = jest.fn();
    const event = {
      preventDefault,
      target: {
        highlights: {
          value: ''
        },
        text: {
          value: ''
        }
      }
    };

    Component.getInstance().handleSubmit(event);

    const text = Component.getInstance().state.text;

    expect(text).toEqual(initState.text);
  });

  it('changes state to non-empty input', () => {
    const Component = renderer.create(<App />);
    const preventDefault = jest.fn();
    const event = {
      preventDefault,
      target: {
        highlights: {
          value: ''
        },
        text: {
          value: 'new text'
        }
      }
    };

    Component.getInstance().handleSubmit(event);

    const text = Component.getInstance().state.text;

    expect(text).toEqual('new text');
  });
});
