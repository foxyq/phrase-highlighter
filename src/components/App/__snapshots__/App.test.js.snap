// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="App"
>
  <section>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="text"
      >
        Sentence or a paragraph to be highlighted
      </label>
      <br />
      <textarea
        cols="70"
        name="text"
        placeholder="You will deliver new technology with an adorable puppy. Perfect!"
        rows="3"
      />
      <label
        htmlFor="highlights"
      >
        Array of highlights objects
      </label>
      <br />
      <textarea
        cols="70"
        name="highlights"
        placeholder="[{\\"startOffset\\":4,\\"endOffset\\":31,\\"color\\":\\"#d9f593\\",\\"priority\\":2},{\\"startOffset\\":1,\\"endOffset\\":9,\\"color\\":\\"pink\\",\\"priority\\":0},{\\"startOffset\\":17,\\"endOffset\\":20,\\"color\\":\\"#a8a8a8\\",\\"priority\\":0},{\\"startOffset\\":37,\\"endOffset\\":55,\\"color\\":\\"orange\\",\\"priority\\":1},{\\"startOffset\\":40,\\"endOffset\\":48,\\"color\\":\\"limegreen\\",\\"priority\\":0},{\\"startOffset\\":55,\\"endOffset\\":64,\\"color\\":\\"yellow\\",\\"priority\\":1}]"
        rows="10"
      />
      <pre
        name="example"
      >
        [{startOffset: 4, endOffset: 31, color: '#d9f593', priority: 2}]
      </pre>
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </section>
  <section>
    
  </section>
</div>
`;
